src/controller/CommissioningWindowOpener.cpp

```diff
@@ -124,6 +136,36 @@ CHIP_ERROR CommissioningWindowOpener::OpenCommissioningWindow(NodeId deviceId, S
     return mController->GetConnectedDevice(mNodeId, &mDeviceConnected, &mDeviceConnectionFailure);
 }
 
+CHIP_ERROR CommissioningWindowOpener::OpenCommissioningWindow(const CommissioningWindowVerifierParams & params)
+{
+    VerifyOrReturnError(mNextStep == Step::kAcceptCommissioningStart, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(kSpake2p_Min_PBKDF_Iterations <= params.GetIteration() &&
+                            params.GetIteration() <= kSpake2p_Max_PBKDF_Iterations,
+                        CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(params.GetSalt().size() >= kSpake2p_Min_PBKDF_Salt_Length &&
+                            params.GetSalt().size() <= kSpake2p_Max_PBKDF_Salt_Length,
+                        CHIP_ERROR_INVALID_ARGUMENT);
+    memcpy(mPBKDFSaltBuffer, params.GetSalt().data(), params.GetSalt().size());
+    mPBKDFSalt = ByteSpan(mPBKDFSaltBuffer, params.GetSalt().size());
+
+    ReturnErrorOnFailure(mVerifier.Deserialize(params.GetVerifier()));
+    mCommissioningWindowVerifierCallback = params.GetCallback();
+    mBasicCommissioningWindowCallback    = nullptr;
+    mCommissioningWindowCallback         = nullptr;
+    mNodeId                              = params.GetNodeId();
+    mCommissioningWindowTimeout          = params.GetTimeout();
+    mPBKDFIterations                     = params.GetIteration();
+    mCommissioningWindowOption           = CommissioningWindowOption::kTokenWithProvidedPIN;
+    mSetupPayload                        = SetupPayload();
+    mSetupPayload.version                = 0;
+    mSetupPayload.discriminator.SetLongValue(params.GetDiscriminator());
+    mSetupPayload.rendezvousInformation.SetValue(RendezvousInformationFlag::kOnNetwork);
```

Comment:
This part is a bit confusing, because in fact the only part of mSetupPayload used in this mode is `mSetupPayload.discriminator`.  So why are we setting its `rendezvousInformation` at all?

Even better would be if we just stored the discriminator outside the setup payload, used that in `OpenCommissioningWindowInternal`, and did not touch the setup payload at all in this codepath.

Reply:
Good idea! Tracking discriminator out of setup payload now and using that for the internal function. The value will still be copied for setup payload for the paths that propagate it via out param or callback (passcode one): https://github.com/project-chip/connectedhomeip/pull/34209/commits/da28dd643e7419092f7a073ab2011fda251f2504